### PR TITLE
Migrate IPvlan docs from CLI to docs repository

### PR DESCRIPTION
**Pull request by thaJeztah**
This migrates the IPVlan docs from the docker/cli repository to this repository, preserving commit history.

Using the following steps;

```console
# install filter-repo (https://github.com/newren/git-filter-repo/blob/main/INSTALL.md)
brew install git-filter-repo

# create a temporary clone of the cli repo
cd ~/projects
git clone https://github.com/docker/cli.git cli_temp
cd cli_temp

# remove all code, except for the files we need, and rename the files
git filter-repo \
    --path-match experimental/vlan-networks.md \
    --path-match experimental/images/ \
    --path-rename experimental/vlan-networks.md:network/ipvlan.md \
    --path-rename experimental/images/ipvlan-l3.gliffy:network/images/ipvlan-l3.gliffy \
    --path-rename experimental/images/ipvlan-l3.png:network/images/ipvlan-l3.png \
    --path-rename experimental/images/ipvlan-l3.svg:network/images/ipvlan-l3.svg \
    --path-rename experimental/images/ipvlan_l2_simple.gliffy:network/images/ipvlan_l2_simple.gliffy \
    --path-rename experimental/images/ipvlan_l2_simple.png:network/images/ipvlan_l2_simple.png \
    --path-rename experimental/images/ipvlan_l2_simple.svg:network/images/ipvlan_l2_simple.svg \
    --path-rename experimental/images/macvlan-bridge-ipvlan-l2.gliffy:network/images/macvlan-bridge-ipvlan-l2.gliffy \
    --path-rename experimental/images/macvlan-bridge-ipvlan-l2.png:network/images/macvlan-bridge-ipvlan-l2.png \
    --path-rename experimental/images/macvlan-bridge-ipvlan-l2.svg:network/images/macvlan-bridge-ipvlan-l2.svg \
    --path-rename experimental/images/multi_tenant_8021q_vlans.gliffy:network/images/multi_tenant_8021q_vlans.gliffy \
    --path-rename experimental/images/multi_tenant_8021q_vlans.png:network/images/multi_tenant_8021q_vlans.png \
    --path-rename experimental/images/multi_tenant_8021q_vlans.svg:network/images/multi_tenant_8021q_vlans.svg \
    --path-rename experimental/images/vlans-deeper-look.gliffy:network/images/vlans-deeper-look.gliffy \
    --path-rename experimental/images/vlans-deeper-look.png:network/images/vlans-deeper-look.png \
    --path-rename experimental/images/vlans-deeper-look.svg:network/images/vlans-deeper-look.svg

# integrate the files in the docs repository
cd ~/projects/docker.github.io
git remote add cli_temp ~/projects/cli_temp
git fetch cli_temp

git merge --allow-unrelated-histories --signoff -S cli_temp/master
```

### CODE REVIEW

**Comment by thaJeztah** [comment]
@usha-mandya @nigelpoulton ptal

File: network/ipvlan.md [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+Ethernet interface or sub-interface to enforce separation between networks and
+connectivity to the physical network.
+
+IPvlan offers a number of unique features and plenty of room for further
+innovations with the various modes. Two high level advantages of these approaches
+are, the positive performance implications of bypassing the Linux bridge and the
+simplicity of having fewer moving parts. Removing the bridge that traditionally
+resides in between the Docker host NIC and container interface leaves a simple
+setup consisting of container interfaces, attached directly to the Docker host
+interface. This result is easy access for external facing services as there is
```
Comment (usha-mandya) [inline network/ipvlan.md:25]:
```suggestion
interface. This result is easy to access for external facing services as there is
```

File: network/ipvlan.md [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+gateway the container receives is `192.168.1.1`.
+
+To help understand how this mode interacts with other hosts, the following
+figure shows the same layer 2 segment between two Docker hosts that applies to
+and IPvlan L2 mode.
+
+![Multiple IPvlan Hosts](images/macvlan-bridge-ipvlan-l2.png)
+
+The following will create the exact same network as the network `db_net` created
+prior, with the driver defaults for `--gateway=192.168.1.1` and `-o ipvlan_mode=l2`.
```
Comment (usha-mandya) [inline network/ipvlan.md:90]:
```suggestion
earlier, with the driver defaults for `--gateway=192.168.1.1` and `-o ipvlan_mode=l2`.
```

File: network/ipvlan.md [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+The option to use either existing parent vlan sub-interfaces or let Docker manage
+them enables the user to either completely manage the Linux interfaces and
+networking or let Docker create and delete the Vlan parent sub-interfaces
+(netlink `ip link`) with no effort from the user.
+
+For example: use `eth0.10` to denote a sub-interface of `eth0` tagged with the
+vlan id of `10`. The equivalent `ip link` command would be
+`ip link add link eth0 name eth0.10 type vlan id 10`.
+
+The example creates the vlan tagged networks and then start two containers to
```
Comment (usha-mandya) [inline network/ipvlan.md:185]:
```suggestion
The example creates the vlan tagged networks and then starts two containers to
```

File: network/ipvlan.md [hunk 4]
```diff
@@ -0,0 +1,629 @@
...
+reachable per IPvlan design in order to isolate container namespaces from the
+underlying host.
+
+**Vlan ID 20**
+
+In the first network tagged and isolated by the Docker host, `eth0.20` is the
+parent interface tagged with vlan id `20` specified with `-o parent=eth0.20`.
+Other naming formats can be used, but the links need to be added and deleted
+manually using `ip link` or Linux configuration files. As long as the `-o parent`
+exists anything can be used if compliant with Linux netlink.
```
Comment (usha-mandya) [inline network/ipvlan.md:197]:
```suggestion
exists, anything can be used if it is compliant with Linux netlink.
```

File: network/ipvlan.md [hunk 5]
```diff
@@ -0,0 +1,629 @@
...
+vlan id of `10`. The equivalent `ip link` command would be
+`ip link add link eth0 name eth0.10 type vlan id 10`.
+
+The example creates the vlan tagged networks and then start two containers to
+test connectivity between containers. Different Vlans cannot ping one another
+without a router routing between the two networks. The default namespace is not
+reachable per IPvlan design in order to isolate container namespaces from the
+underlying host.
+
+**Vlan ID 20**
```
Comment (usha-mandya) [inline network/ipvlan.md:191]:
```suggestion
### Vlan ID 20
```

File: network/ipvlan.md [hunk 6]
```diff
@@ -0,0 +1,629 @@
...
+    --subnet=192.168.20.0/24 \
+    --gateway=192.168.20.1 \
+    -o parent=eth0.20 ipvlan20
+
+# in two separate terminals, start a Docker container and the containers can now ping one another.
+$ docker run --net=ipvlan20 -it --name ivlan_test1 --rm alpine /bin/sh
+$ docker run --net=ipvlan20 -it --name ivlan_test2 --rm alpine /bin/sh
+```
+
+**Vlan ID 30**
```
Comment (usha-mandya) [inline network/ipvlan.md:211]:
```suggestion
### VLAN ID 30**
```

File: network/ipvlan.md [hunk 7]
```diff
@@ -0,0 +1,629 @@
...
+for 802.1q trunks with VLAN IDs each network gets mapped to the corresponding
+VLAN/Subnet from the network.
+
+An example being, NetOps provides VLAN ID and the associated subnets for VLANs
+being passed on the Ethernet link to the Docker host server. Those values are
+simply plugged into the `docker network create` commands when provisioning the
+Docker networks. These are persistent configurations that are applied every time
+the Docker engine starts which alleviates having to manage often complex
+configuration files. The network interfaces can also be managed manually by
+being pre-created and docker networking will never modify them, simply use them
```
Comment (usha-mandya) [inline network/ipvlan.md:274]:
```suggestion
being pre-created and Docker networking will never modify them, simply use them
```

File: network/ipvlan.md [hunk 8]
```diff
@@ -0,0 +1,629 @@
...
+
+## IPvlan L3 mode example
+
+IPvlan will require routes to be distributed to each endpoint. The driver only
+builds the IPvlan L3 mode port and attaches the container to the interface. Route
+distribution throughout a cluster is beyond the initial implementation of this
+single host scoped driver. In L3 mode, the Docker host is very similar to a
+router starting new networks in the container. They are on networks that the
+upstream network will not know about without route distribution. For those
+curious how IPvlan L3 will fit into container networking see the following
```
Comment (usha-mandya) [inline network/ipvlan.md:293]:
```suggestion
curious how IPvlan L3 will fit into container networking, see the following
```

File: network/ipvlan.md [hunk 9]
```diff
@@ -0,0 +1,629 @@
...
+> # Inside an L3 mode container
+> $$ ip route
+>  default dev eth0
+>   192.168.214.0/24 dev eth0  src 192.168.214.10
+> ```
+
+In order to ping the containers from a remote Docker host or the container be
+able to ping a remote host, the remote host or the physical network in between
+need to have a route pointing to the host IP address of the container's Docker
+host eth interface. More on this as we evolve the IPvlan `L3` story.
```
Comment (usha-mandya) [inline network/ipvlan.md:375]: > More on this as we evolve the IPvlan `L3` story.

Do we need this?

File: network/ipvlan.md [hunk 10]
```diff
@@ -0,0 +1,629 @@
...
+> same v6 address it throws the following like the address isn't properly being
+> released to the v6 pool. It will fail to unmount the container and be left dead.
+
+```console
+docker: Error response from daemon: Address already in use.
+```
+
+## Manually create 802.1q links
+
+**Vlan ID 40**
```
Comment (usha-mandya) [inline network/ipvlan.md:573]:
```suggestion
### VLAN ID 40
```

File: network/ipvlan.md [hunk 11]
```diff
@@ -0,0 +1,629 @@
...
+created when you call a `docker network create`. `docker network rm` will delete
+the sub-interface. Parent interfaces such as `eth0` are not deleted, only
+sub-interfaces with a netlink parent index > 0.
+
+For the driver to add/delete the vlan sub-interfaces the format needs to be
+`interface_name.vlan_tag`. Other sub-interface naming can be used as the
+specified parent, but the link will not be deleted automatically when
+`docker network rm` is invoked. 
+
+The option to use either existing parent vlan sub-interfaces or let Docker manage
```
Comment (usha-mandya) [inline network/ipvlan.md:176]: We've used both `vlan` and `VLAN` in this doc. Suggest we use VLAN throughout for consistency,

**Comment by usha-mandya** [comment]
@thaJeztah Thanks for the PR. Made some minor suggestions. The PR looks good otherwise.

**Review by usha-mandya — APPROVED** [review]
LGTM

**Pull request merged**